src/vs/editor/contrib/stickyScroll/browser/stickyScroll.ts

```diff
@@ -66,12 +67,31 @@ class StickyScrollController extends Disposable implements IEditorContribution {
 			this._editor.addOverlayWidget(this.stickyScrollWidget);
 			this._sessionStore.add(this._editor.onDidChangeModel(() => this._update(true)));
 			this._sessionStore.add(this._editor.onDidScrollChange(() => this._update(false)));
+			this._sessionStore.add(this._editor.onDidChangeModelTokens((e) => this._onTokensChange(e)));
 			this._sessionStore.add(this._editor.onDidChangeModelContent(() => this._updateSoon.schedule()));
 			this._sessionStore.add(this._languageFeaturesService.documentSymbolProvider.onDidChange(() => this._update(true)));
 			this._update(true);
 		}
 	}
 
+	private _onTokensChange(event: IModelTokensChangedEvent) {
+		const widgetLineRange = this.stickyScrollWidget.getCurrentLineRange();
+		let rerender: boolean = false;
+		for (const range of event.ranges) {
+			const fromLineNumber = range.fromLineNumber;
+			const toLineNumber = range.toLineNumber;
+			const fromLineWidget = widgetLineRange[0];
+			const toLineWidget = widgetLineRange[1];
+			if (fromLineNumber <= toLineWidget && toLineNumber >= fromLineWidget) {
+				rerender = true;
+				break;
+			}
```

Comment:
This should handle that `widgetLineRange` is a set of N lines, not just the extremes.